be/test/exprs/vectorized/lambda_function_expr_test.cpp

```diff
@@ -0,0 +1,372 @@
+                EXPECT_EQ(1, result->get(1).get_array()[0].get_int8());
+                EXPECT_EQ(1, result->get(1).get_array()[1].get_int8());
+                EXPECT_EQ(1, result->get(2).get_array()[0].get_int8());
+                EXPECT_EQ(0, result->get(2).get_array()[1].get_int8());
+            } else if (i == 1 && j == 2) { // // array_map(x -> x+a, array<int>)
+                EXPECT_EQ(3, result->size());
+                EXPECT_EQ(2, result->get(0).get_array()[0].get_int32());
+                EXPECT_EQ(5, result->get(0).get_array()[1].get_int32());
+                ASSERT_TRUE(result->get(1).get_array()[0].is_null());
+                ASSERT_TRUE(result->get(1).get_array()[1].is_null());
+                ASSERT_TRUE(result->get(2).get_array()[0].is_null());
+                EXPECT_EQ(13, result->get(2).get_array()[1].get_int32());
+            } else if (i == 2 && (j == 0 || j == 2)) { // array_map( x->x || x->x+a, [null])
+                EXPECT_EQ(1, result->size());
+                ASSERT_TRUE(result->get(0).get_array()[0].is_null());
+            } else if (i == 2 && j == 1) { // array_map(x -> x is null,[null])
+                EXPECT_EQ(1, result->size());
+                EXPECT_EQ(1, result->get(0).get_array()[0].get_int8());
+            } else if (i == 3) { // array_map(x->xxx,[])
+                EXPECT_EQ(1, result->size());
+                ASSERT_TRUE(result->get(0).get_array().empty());
+            } else if (i == 4 && (j == 0 || j == 2)) { // array_map(x->x || x->x+a, array<special>)
+                                                       // [null]
+                                                       // []
+                                                       // NULL
+                EXPECT_EQ(3, result->size());
+                ASSERT_TRUE(result->get(0).get_array()[0].is_null());
+                ASSERT_TRUE(result->get(1).get_array().empty());
+                ASSERT_TRUE(result->is_null(2));
+            } else if (i == 4 && j == 1) { // array_map(x->x is null, array<special>)
+                EXPECT_EQ(3, result->size());
+                EXPECT_EQ(1, result->get(0).get_array()[0].get_int8());
+                ASSERT_TRUE(result->get(1).get_array().empty());
+                ASSERT_TRUE(result->is_null(2));
+            } else if (i == 5 && j == 0) { // array_map( x->x, array<const[1,4]...>)
+                EXPECT_EQ(3, result->size());
+                EXPECT_EQ(1, result->get(0).get_array()[0].get_int32());
+                EXPECT_EQ(4, result->get(0).get_array()[1].get_int32());
+                EXPECT_EQ(1, result->get(1).get_array()[0].get_int32());
+                EXPECT_EQ(4, result->get(1).get_array()[1].get_int32());
+                EXPECT_EQ(1, result->get(2).get_array()[0].get_int32());
+                EXPECT_EQ(4, result->get(2).get_array()[1].get_int32());
+            } else if (i == 5 && j == 1) { // array_map(x->x is null, array<const[1,4]...>)
```

Comment:
Why not write them in a table-driven pattern?

Reply:
what do you mean?